compiler/record-minmax/include/RecordMinMax.h

```diff
@@ -23,14 +23,19 @@
 #include "MinMaxObserver.h"
 
 #include <memory>
+#include <thread>
 
 namespace record_minmax
 {
 
 class RecordMinMax
 {
 public:
-  explicit RecordMinMax() = default;
+  explicit RecordMinMax(bool use_parallel)
+    : _threads_size(use_parallel ? std::thread::hardware_concurrency() : 1)
```

Comment:
Better to make the number of threads configurable in the command line option. Perhaps it would be better to replace `--parallel` with `--num_threads <int>` (default: 1).

`std::thread::hardware_concurrency()` can return 0, and it will cause problem in release mode. In addition, I'm not sure the function works as expected in CI machine.